client/features/auth-pages/BackOffice.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState, useEffect } from 'react';
+import { Box, Button } from '@mui/material';
+import DatePicker from 'react-datepicker';
+import style from '@styles/BackOffice.module.css';
+
+import 'react-datepicker/dist/react-datepicker.css';
+
+const BackOffice = () => {
+  const today = new Date();
+  const tomorrowMidnight = new Date(today);
+  tomorrowMidnight.setDate(today.getDate() + 1);
+  const [endDate, setEndDate] = useState(
+    new Date(new Date(tomorrowMidnight).setHours(0, 0, 0, 0)),
+  );
+  const [startDate, setStartDate] = useState(
+    new Date(new Date().setHours(0, 0, 0, 0)),
+  );
+
+  const startDateMs = startDate.getTime();
+  const endDateMs = endDate.getTime();
+
+  // Calculate the difference in milliseconds
+  const differenceMs = Math.abs(endDateMs - startDateMs); // Use Math.abs to get a positive difference
+  // Convert milliseconds to days
+  const millisecondsPerDay = 1000 * 60 * 60 * 24;
+  const differenceDays = Math.floor(differenceMs / millisecondsPerDay);
+
+  useEffect(() => {
+    if (startDate >= endDate) {
+      alert('Start date must be before end date');
+    }
+    if (differenceDays > 30) {
+      alert('Date range must be less than 30 days');
+    }
+  }, [startDate, endDate, differenceDays]);
+
+  const shortDateString = (date: Date) => {
+    return date.toISOString().split('T')[0].slice(2).split('-').join('');
+  };
+
+  const downloadLogs = async () => {
```

Comment:
Since we have an async operation here, I'd highly recommend we incorporate error handling. Also we can use the `axiosClient` to simply the API call.

```typescript
import axiosClient from '@config/axios';

  const downloadLogs = async () => {
    if (differenceDays > 30) {
      alert('Date range must be less than 30 days');
      return;
    }

    try {
      const response = await axiosClient.post(
        '/v1/logs',
        { startDate, endDate },
        { responseType: 'blob' },
      );

      const blob = new Blob([response.data]);
      const url = window.URL.createObjectURL(blob);
      const link = document.createElement('a');
      link.href = url;
      link.setAttribute(
        'download',
        `linkta-logs_${shortDateString(startDate)}_${shortDateString(endDate)}.zip`,
      );
      document.body.appendChild(link);
      link.click();
      link.parentNode?.removeChild(link);
    } catch (error) {
      if (error instanceof Error) {
        alert(`Error: ${error.message}`);
      } else {
        alert('An unexpected error occurred.');
      }
    }
  };
```

Reply:
Done.